deepcell_toolbox/metrics.py

```diff
@@ -278,7 +283,10 @@ def get_box_labels(images):
         y_pred_boxes, y_pred_labels = get_box_labels(self.y_pred.astype('int'))
 
         # has the form [gt_bbox, res_bbox]
-        overlaps = compute_overlap(y_true_boxes, y_pred_boxes)
+        if self.is_3d:
+            overlaps = compute_overlap_3D(y_true_boxes, y_pred_boxes)
+        else:
+            overlaps = compute_overlap(y_true_boxes, y_pred_boxes)
```

Comment:
Is there a way we can infer this from the data itself? It would be nice to not require another flag if we don't have to.

Reply:
I don't know if there's a good way to do this, since data needs to be able to both have channels and not